sdk/samples/iot/pnp/pnp_thermostat_component.h

```diff
@@ -17,47 +17,118 @@
+  bool send_maximum_temperature_property;
 } pnp_thermostat_component;
 
+/*
+ * @brief Initialize a #pnp_thermostat_component which holds device thermostat info.
+ *
+ * @param[out] out_thermostat_component A pointer to a #out_thermostat_component instance to
+ *                                      initialize.
+ * @param[in] component_name The name of the component.
+ * @param[in] initial_temperature The initial temperature to set all temperature member variables.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK #pnp_thermostat_component is initialied successfully.
+ * @retval #AZ_ERROR_ARG The pointer to the #pnp_thermostat_component instance is NULL.
+ */
 az_result pnp_thermostat_init(
-    pnp_thermostat_component* thermostat_component,
+    pnp_thermostat_component* out_thermostat_component,
     az_span component_name,
-    double initial_temp);
-
-az_result pnp_thermostat_get_telemetry_message(
-    const az_iot_hub_client* client,
-    const pnp_thermostat_component* thermostat_component,
-    pnp_mqtt_message* mqtt_message);
+    double initial_temperature);
 
-bool pnp_thermostat_get_max_temp_report(
-    const az_iot_hub_client* client,
+/**
+ * @brief Build the thermostat's temperature telemetry message.
+ *
+ * @param[in] thermostat_component A pointer to the themostat component to get data.
+ * @param[out] payload An #az_span with sufficient capacity to hold the json payload.
+ * @param[out] out_payload A pointer to the #az_span containing the output json payload.
+ */
+void pnp_thermostat_build_telemetry_message(
     pnp_thermostat_component* thermostat_component,
-    pnp_mqtt_message* mqtt_message);
+    az_span payload,
+    az_span* out_payload);
 
-az_result pnp_thermostat_process_property_update(
-    const az_iot_hub_client* client,
+/**
+ * @brief Build the thermostat's maximum temperature reported property message.
+ *
+ * @param[in] thermostat_component A pointer to the themostat component to get data.
+ * @param[out] out_property_name The name of the reported property to be sent.
+ * @param[out] payload An #az_span with sufficient capacity to hold the json payload.
+ * @param[out] out_payload A pointer to the #az_span containing the output json payload.
+ */
+void pnp_thermostat_build_maximum_temperature_reported_property(
     pnp_thermostat_component* thermostat_component,
+    az_span* out_property_name,
+    az_span payload,
+    az_span* out_payload);
+
+/**
+ * @brief Build the thermostat's error message with status.
+ *
+ * @param[in] component_name The name of the component for the reported property.
+ * @param[in] property_name The name of the property for which to send an update.
+ * @param[in] property_value The property value to be appended.
+ * @param[in] status The return status for the error message ack.
+ * @param[in] version The version for the reported property ack.
+ * @param[out] payload An #az_span with sufficient capacity to hold the json payload.
```

Comment:
thanks I forgot to change these back.

Reply:
wait, nm above comment.  I confused [out[ with out_ prefix.  

@antkmsft what is your thought on this?  I have [out] set in another sample file like so:
```json
/*
 * @brief Generate the base64 encoded and signed signature using HMAC-SHA256 signing.
 *
 * @param[in] sas_base64_encoded_key An #az_span containing the SAS key that will be used for
 *                                   signing.
 * @param[in] sas_signature An #az_span containing the signature.
 * @param[out] sas_base64_encoded_signed_signature An #az_span with sufficient capacity to hold the
 *                                                 encoded signed signature.
 * @param[out] out_sas_base64_encoded_signed_signature A pointer to the #az_span containing the
 *                                                     encoded signed signature.
 */
void iot_sample_generate_sas_base64_encoded_signed_signature(
    az_span sas_base64_encoded_key,
    az_span sas_signature,
    az_span sas_base64_encoded_signed_signature,
    az_span* out_sas_base64_encoded_signed_signature);
```

Reply:
The `[out]` should be changed to `[in]` because that span (without the `out_` prefix) tells the API what pointer and how much space it has to use. It isn't referenced again by the user after the function call for any information.